### PR TITLE
Proposed Fix: #17964 sprites don't show up on two sides of the view-clip selection

### PR DESCRIPTION
**Pull request by ebstalk**
Hello, this is my first contribution so any feedback is welcomed.

This is a screenshot from the current version, there are peeps here that arent shown.
Current version

My proposed fixed code
Proposed fix version

I feel like this is just a bandaid fix to another issue because it seems that  `gClipSelectionA` and `gClipSelectionB` have the same location, hence the added 28 to get the other corner of the tile. I'm not sure if this is an acceptable fix because of this.

### CODE REVIEW

**Comment by Gymnasiast** [comment]
It seems you are on the right track, but the value we need is `COORDS_XY_STEP - 1` (= 31). That ensures the sprites on the whole tile are taken in.

You’ll also need to add a changelog entry. You simply copy the bug description from the original issue and put it in the correct place (i.e. the Fix section, in order), making sure to end the sentence with a dot.

**Comment by ebstalk** [comment]
Thank you for responding, is there anything else I need to add?

File: src/openrct2/paint/Paint.Entity.cpp [hunk 1]
```diff
@@ -85,11 +85,11 @@ void EntityPaintSetup(paint_session& session, const CoordsXY& pos)
             {
                 continue;
             }
-            if (entityPos.x < gClipSelectionA.x || entityPos.x > gClipSelectionB.x)
+            if (entityPos.x < gClipSelectionA.x || entityPos.x > gClipSelectionB.x + 31)
```
Comment (Gymnasiast) [inline src/openrct2/paint/Paint.Entity.cpp:88]: No naked numbers, please. As I said, please use `COORDS_XY_STEP - 1`.

File: distribution/changelog.txt [hunk 2]
```diff
@@ -74,6 +74,7 @@
 - Fix: [#17980] Queue lines of track designs mess up existing queue lines if dragged through them.
 - Fix: [#17959] Areas marked for dirty drawing are too large.
 - Fix: [#17963] Some marketing campaigns can’t be started after Finances window tab has been on Research.
+- Fix: [#17964] Sprites don't show up on two sides of the view-clip selection.
```
Comment (Gymnasiast) [inline distribution/changelog.txt:77]:
```suggestion
- Fix: [#17964] Sprites don’t show up on two sides of the view-clip selection.
```

(Typewriter quotes are for code ;-) )

**Comment by Gymnasiast** [comment]
You need to address the line notes, but once you do that, it’s good to go.

**Comment by Broxzier** [comment]
> I feel like this is just a bandaid fix to another issue because it seems that gClipSelectionA and gClipSelectionB have the same location

That's a good intuition. The main problem here is that the selection is tile based, while the entities' positions are not.

File: src/openrct2/paint/Paint.Entity.cpp (hunk 1 above)
Comment (Broxzier) [reply src/openrct2/paint/Paint.Entity.cpp:88]: I think it would be cleaner to use `>=` here and leave out the `- 1`. The same applies to the line below.

Comment (ebstalk) [reply src/openrct2/paint/Paint.Entity.cpp]: My bad I misunderstood what you were trying to say with the numbers. I have applied both changes.

**Comment by ebstalk** [comment]
Thank yall for the feedback!


> > I feel like this is just a bandaid fix to another issue because it seems that gClipSelectionA and gClipSelectionB have the same location
> 
> That's a good intuition. The main problem here is that the selection is tile based, while the entities' positions are not.

Will this fix be sufficient then?

**Comment by Gymnasiast** [comment]
> > I feel like this is just a bandaid fix to another issue because it seems that gClipSelectionA and gClipSelectionB have the same location
> 
> That's a good intuition. The main problem here is that the selection is tile based, while the entities' positions are not.

I’m not sure I agree, I think this is about as clean as you realistically expect.

File: src/openrct2/paint/Paint.Entity.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/paint/Paint.Entity.cpp]: I disagree, `COORDS_XY_STEP - 1` makes it much clearer it’s confined to this tile.

**Review by Gymnasiast — APPROVED** [review]

**Comment by Broxzier** [comment]
Not sure how you can disagree with a fact? Just check the A and B coordinates when selecting a single tile: they are the same. Do you mean you disagree with how this currently works?

**Comment by Gymnasiast** [comment]
I was disagreeing with the notion it was a band-aid fix. I though the second half of my sentence would have provided that context.

**Comment by Broxzier** [comment]
I see, quoting my response made it confusing then.

**Pull request merged**